src/validation.h

```diff
@@ -164,8 +164,11 @@ bool LoadGenesisBlock(const CChainParams& chainparams);
 void UnloadBlockIndex();
 /** Run an instance of the script checking thread */
 void ThreadScriptCheck(int worker_num);
-/** Retrieve a transaction (from memory pool, or from disk, if possible) */
-bool GetTransaction(const uint256& hash, CTransactionRef& tx, const Consensus::Params& params, uint256& hashBlock, const CBlockIndex* const blockIndex = nullptr);
+/**
+ * Return transaction in txOut, and if it was found via block_index, its hash is placed in hashBlock.
```

Comment:
Thanks for consolidating the function comment to the header declaration.

Would it be too much to ask to doxygenize this, specifically to show which arguments are required in args (hash, params), which are optional in args (mempool, block_index), and which are out args (txOut, hashBlock). Even better would be to re-order them to required in args, optional in args, out args.

Reply:
thx, doxygenized :ox: